dart/dynamics/Shape.h

```diff
@@ -191,29 +142,19 @@ class Shape : public virtual common::Subject
   /// Instruct this shape to update its data
   virtual void refreshData();
 
-  /// \brief
-  virtual void draw(renderer::RenderInterface* _ri = nullptr,
-                    const Eigen::Vector4d& _color = Eigen::Vector4d::Ones(),
-                    bool _useDefaultColor = true) const = 0;
-
-  /// Pass in true to prevent this shape from being rendered. Pass in false to
-  /// allow it to render again.
-  void setHidden(bool _hide=true);
+  /// Notify that the color (rgba) of this shape has updated
+  virtual void notifyColorUpdate(const Eigen::Vector4d& color);
```

Comment:
I don't know if this would be silly or not, but we could potentially make VisualAddon be an AddonManager and be able to attach a MeshColor Addon which could contain per-vertex color information.

There might be a better way to approach it than what I've suggested, but I think there would be value in being able to decouple the geometric and visual data. For example, suppose you want to be able to visualize the same geometric data with different color overlays (say you have two copies of the same robot, but you want different per-vertex coloring on each to highlight different contextual data). If the coloring is tied into the geometric data, that's a significant amount of wasted memory, because you'll need a full copy.